### ISSUE TITLE
Allow custom HTML value

### ISSUE THREAD

**Issue by rooneyg21** [sindresorhus/linkify-urls#10]
This would help solve sindresorhus/refined-github#703

```js
const linkify = (href, options) => createHtmlElement({
	name: 'a',
	attributes: Object.assign({href: ''}, options.attributes, {href}),
	value: options.value || href
}); 
```

If the user can pass the value as part of the options, the library can parse the href text and add an anchor element that has whatever the user wants as its text (HTML in the case of sindresorhus/refined-github#695 to preserve the span tags)

**Comment by sindresorhus** [comment]
Sure. Pull request welcome.

**Comment by yeskunall** [comment]
Wait, I'm confused. How will adding a check for `options.value` fix the issue of parsing the whole URL?

```js
createHtmlElement({
    name: 'h1',
    attributes: {
        class: 'unicorn',
        rainbow: true,
        horse: false,
        number: 1,
        multiple: ['a', 'b']
    },
    value: '🦄'
});
//=> '<h1 class="unicorn" rainbow number="1" multiple="a b">🦄</h1>'
```

Care to explain what you mean @arungalva?

**Comment by fregante** [comment]
Sadly this isn't useful if `linkify-urls` is used to to match URLs inside random strings.

https://runkit.com/embed/rys5wsu5zn99

If we don't know what part of `input` is a URL, when does a static `value` ever come useful? It should be a callback on each matched URL.

Related: https://github.com/sindresorhus/refined-github/issues/811#issuecomment-350296939

**Comment by sindresorhus** [comment]
Fixed by #18
